### DISCUSSION TITLE
Allow for folder mapping notes in the root folder

### DISCUSSION THREAD

**Discussion by chrisgurney** [chrisgurney/obsidian-note-toolbar#27] [Ideas]
From [this thread](https://www.reddit.com/r/ObsidianMD/comments/1bzwk24/comment/kysungw/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button) from Reddit user [abhuva79](https://www.reddit.com/user/abhuva79/):
> When setting a folder to show a toolbar - this seems to not work on the base folder "/"
> ...
> As the majority of my notes lives in the base-folder, i would love to either get told what i did wrong, or the (possible) bug fixed =)

**Comment by chrisgurney** [comment]
I think the glaring missing thing is the ability to map to all items in the vault (i.e., support for: `*`). If the user needs any specific folders in lower-level folders, they'd just create mappings for those (to override).

I'm working on a beta with some other items at the moment, and will include this enhancement in that release.

**Reply by chrisgurney** [reply]
**Related Issue: Preference to _not_ show toolbars in subfolders**

It hasn't come up yet, but this solution doesn't solve the use case of wanting to map toolbars to items in the root (or any folder for that matter), but *not* in subfolders. For that to work, a solution might consist of:

- A flag as to whether a particular mapping applies to subfolders or not.
- A character (such as `/`) that allows for mappings specific to the root only.

**Comment by 23426356587** [comment]
Can confirm - this happened to me and I had to add a rule with the same toolbar for the first layer of folders in my vault.

**Comment by chrisgurney** [comment]
New folder mappings are new available [in the beta](https://github.com/chrisgurney/obsidian-note-toolbar/discussions/40):

- `/` = map toolbars for notes that just live in the root folder
- `*` = apply toolbars for all of your notes (note that you can still override this with more specific folder mappings or the property)
  - Tip: Put this at the bottom of your Folder Mappings in order to make sure more specific folder mappings take precedence.

[Read more about the beta.](https://github.com/chrisgurney/obsidian-note-toolbar/discussions/40)

**Comment by chrisgurney** [comment]
This feature has now been [released](https://github.com/chrisgurney/obsidian-note-toolbar/discussions/40).
